sdk/modelsrepository/Azure.Iot.ModelsRepository/src/RepositoryHandler.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Iot.ModelsRepository.Fetchers;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Iot.ModelsRepository
+{
+    internal class RepositoryHandler
+    {
+        private readonly IModelFetcher _modelFetcher;
+        private readonly Guid _clientId;
+        private readonly ClientDiagnostics _clientDiagnostics;
+        private readonly Uri _repositoryUri;
+        private readonly ModelsRepositoryClientOptions _clientOptions;
+
+        public RepositoryHandler(Uri repositoryUri, ClientDiagnostics clientDiagnostics, ModelsRepositoryClientOptions options)
+        {
+            Argument.AssertNotNull(options, nameof(options));
+
+            _clientOptions = options;
+            _clientDiagnostics = clientDiagnostics;
+            _modelFetcher = repositoryUri.Scheme == "file"
```

Comment:
Should "file" be a constant somewhere?

Reply:
it's never reused, but sure, we can have it in the constants file.